edx_rest_framework_extensions/authn/jwt/middleware.py

```diff
@@ -0,0 +1,70 @@
+"""
+Middleware supporting JWT Authentication.
+"""
+import logging
+
+from edx_django_utils import monitoring
+
+from edx_rest_framework_extensions.authn.jwt.cookies import (
+    jwt_cookie_name,
+    jwt_cookie_header_payload_name,
+    jwt_cookie_signature_name,
+)
+from edx_rest_framework_extensions.authn.jwt.constants import JWT_DELIMITER
+
+log = logging.getLogger(__name__)
+
+
+class JwtAuthCookieMiddleware(object):
+    """
+    Reconstitutes JWT auth cookies for use by API views which use the JwtAuthentication
+    authentication class.
+
+    We split the JWT across two separate cookies in the browser for security reasons. This
+    middleware reconstitutes the full JWT into a new cookie on the request object for use
+    by the JwtAuthentication class.
```

Comment:
(Can be addressed in a subsequent PR.)
Nit: Link to the edx-platform decision record?

Reply:
Agreed.  Should be updated.